geode-connectors/src/test/java/org/apache/geode/connectors/jdbc/internal/cli/DestroyDataSourceCommandTest.java

```diff
@@ -0,0 +1,267 @@
+
+  @Before
+  public void setUp() throws Exception {
+    cache = mock(InternalCache.class);
+    command = spy(DestroyDataSourceCommand.class);
+    doReturn(cache).when(command).getCache();
+    cacheConfig = mock(CacheConfig.class);
+    ccService = mock(InternalConfigurationPersistenceService.class);
+
+    doReturn(Collections.emptySet()).when(command).findMembers(any(), any());
+    doReturn(ccService).when(command).getConfigurationPersistenceService();
+    when(ccService.getCacheConfig(any())).thenReturn(cacheConfig);
+    doAnswer(invocation -> {
+      UnaryOperator<CacheConfig> mutator = invocation.getArgument(1);
+      mutator.apply(cacheConfig);
+      return null;
+    }).when(ccService).updateCacheConfig(any(), any());
+
+    when(ccService.getConfigurationRegion()).thenReturn(mock(Region.class));
+    when(ccService.getConfiguration(any())).thenReturn(mock(Configuration.class));
+  }
+
+  @Test
+  public void missingMandatory() {
+    gfsh.executeAndAssertThat(command, COMMAND).statusIsError().containsOutput("Invalid command");
+  }
+
+  @Test
+  public void returnsErrorIfBindingDoesNotExistAndIfExistsUnspecified() {
+    gfsh.executeAndAssertThat(command, COMMAND + " --name=name").statusIsError()
+        .containsOutput("does not exist.");
+  }
+
+  @Test
+  public void skipsIfBindingDoesNotExistAndIfExistsSpecified() {
+    gfsh.executeAndAssertThat(command, COMMAND + " --name=name --if-exists").statusIsSuccess()
+        .containsOutput("does not exist.");
+  }
+
+  @Test
+  public void skipsIfBindingDoesNotExistAndIfExistsSpecifiedTrue() {
+    gfsh.executeAndAssertThat(command, COMMAND + " --name=name --if-exists=true").statusIsSuccess()
+        .containsOutput("does not exist.");
+  }
+
+  @Test
+  public void returnsErrorIfBindingDoesNotExistAndIfExistsSpecifiedFalse() {
+    gfsh.executeAndAssertThat(command, COMMAND + " --name=name --if-exists=false").statusIsError()
+        .containsOutput("does not exist.");
+  }
+
+  @Test
+  public void whenNoMembersFoundAndNoClusterConfigServiceRunningThenError() {
+    doReturn(Collections.emptySet()).when(command).findMembers(any(), any());
+    doReturn(null).when(command).getConfigurationPersistenceService();
+
+    gfsh.executeAndAssertThat(command, COMMAND + " --name=name").statusIsError()
+        .containsOutput("No members found and cluster configuration disabled.");
+  }
+
+  @Test
+  public void whenNoMembersFoundAndClusterConfigRunningThenUpdateClusterConfig() {
+    List<JndiBindingsType.JndiBinding> bindings = new ArrayList<>();
+    JndiBindingsType.JndiBinding jndiBinding = new JndiBindingsType.JndiBinding();
+    jndiBinding.setJndiName("name");
+    bindings.add(jndiBinding);
+    doReturn(bindings).when(cacheConfig).getJndiBindings();
+
+    gfsh.executeAndAssertThat(command, COMMAND + " --name=name").statusIsSuccess()
+        .containsOutput("No members found.")
+        .containsOutput("Changes to configuration for group 'cluster' are persisted.");
```

Comment:
The "Change" message is a recent addition of the management team. It is not part of this pull request. But what we can do is change our implementation of updateConfigForGroup to return false in this case instead of true.

Reply:
After looking over this, there is no code path which would allow us to attempt updating cluster config in the event the binding is not found within cluster config.

@jchen21 The reason you might see those two messages together is in the event that there are no servers but cluster config is enabled and has the specified binding in it's list of bindings. In this case the command would update cluster config to remove the binding, but would also return 'no members found' as it wouldn't be able to construct a table of function results (the normal output for a successful destroy command) since no servers were available to run the function.